packages/react-components/src/components/Tooltip/Tooltip.spec.tsx

```diff
@@ -201,4 +201,40 @@ describe('<Tooltip> component', () => {
 
     cleanup();
   });
+
+  it('should not call onOpen callback redundantly when hovering from trigger element to tooltip', () => {
+    const onOpen = vi.fn();
+    const onClose = vi.fn();
+    const { getByRole, queryByText } = render(
+      <Tooltip
+        triggerRenderer={() => <button>Open</button>}
+        triggerOnClick={false}
+        withFadeAnimation={false}
+        hoverOutDelayTimeout={0}
+        transitionDuration={0}
+        onOpen={onOpen}
+        onClose={onClose}
+      >
+        <div style={{ width: '100px' }}>Test</div>
+      </Tooltip>
+    );
+
+    const button = getByRole('button', { name: 'Open' });
+
+    act(() => {
+      fireEvent.mouseEnter(button);
+    });
+
+    expect(queryByText('Test')).toBeInTheDocument();
+
+    act(() => {
+      fireEvent.mouseEnter(queryByText('Test') as HTMLElement);
+      fireEvent.mouseEnter(button);
```

Comment:
before this fix, these hovers were causing additional calls of `onOpen` callback